doc/design.md

```diff
@@ -157,10 +157,59 @@ Returns: auth token
 - `400 BAD FORMAT`: missing email or password
 
 
+**`GET`** `/api/users/`
+
+Fetches the list of existing users
+
+Required headers:
+
+- `Authorization: <JWT string>`
+
+Returns: List of users:
+
+- format: `JSON`
+- fields:
+    * users :: list
+        * email
+        * admin
+        * timestamp
+
+Status codes:
+
+- `200 OK`
+- `401 UNAUTHORIZED`: authorization rejected; missing or invalid auth token
+- `403 FORBIDDEN`: authorized user is not admin
+
+
+**`GET`** `/api/users/EMAIL`
+
```

Comment:
Using the user's email address as the unique identifier for lookup in  a URL feels weird (as a login name it's fine). I'd rather see a user id or username here. Not adamantly opposed, just want to discuss it before it's implemented

Reply:
Yea, I agree it's weird and think it's worth revisiting. This particular endpoint isn't implemented in this PR, and I'm also unsure how we would even use this, i.e. what is the frontend component that would fetch only a single user's info? Apart from when that user is logging/ logged in, in which case the login endpoint provides this information.

Reply:
I agree with @erikjasso as I'm not opposed to it, but I believe its common practice to use a user id for this action.

Reply:
In our spec, email is the only public-facing identifier for a user because we eliminated the "username" field, deciding it was functionally redundant with just using an email address, and the user_id field is a UUID generated by the database. 

I can pull this design question into a new issue since it's not in this PR

Reply:
I have seen this done as an `/api/users/search` as a **post** where you send the required information along in the request ie: Email. That way you don't have to expose the UUID to the front end.

I agree that it could be beneficial to spec out naming conventions and requisite data for the endpoints at our next meeting.